applications/OptimizationApplication/python_scripts/model_part_controllers/connectivity_preserving_model_part_controller.py

```diff
@@ -0,0 +1,65 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.model_part_controllers.model_part_controller import ModelPartController
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, _) -> ModelPartController:
+    if not parameters.Has("settings"):
+        raise RuntimeError(f"ConnectivityPreservingModelPartController instantiation requires a \"settings\" in parameters [ parameters = {parameters}].")
+    return ConnectivityPreservingModelPartController(model, parameters["settings"])
+
+class ConnectivityPreservingModelPartController(ModelPartController):
+    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters):
+        default_settings = Kratos.Parameters("""{
+            "transformation_settings": [
+                {
+                    "source_model_part_name"     : "",
+                    "destination_model_part_name": "",
+                    "destination_element_name"   : "",
+                    "destination_condition_name" : ""
+                }
+            ]
+        }""")
+
+        self.model = model
+        self.parameters = parameters
+        self.parameters.ValidateAndAssignDefaults(default_settings)
+
+        for sub_model_part_settings in self.parameters["transformation_settings"].values():
+            sub_model_part_settings.ValidateAndAssignDefaults(default_settings["transformation_settings"].values()[0])
+
+    def ImportModelPart(self) -> None:
+        connectivity_preserve_modeller = Kratos.ConnectivityPreserveModeler()
+        for transformation_settings in self.parameters["transformation_settings"].values():
+            # get source and destination model parts
+            source_model_part = self.model[transformation_settings["source_model_part_name"].GetString()]
+            destination_model_part_name: str = transformation_settings["destination_model_part_name"].GetString()
+            if not self.model.HasModelPart(destination_model_part_name):
+                destination_model_part = self.model.CreateModelPart(destination_model_part_name)
+            else:
+                destination_model_part = self.model[destination_model_part_name]
```

Comment:
if the model_part exists, should we check that it is empty??

Reply:
IF it exists, `ConnectivityPreserModeller` will remove all the entities from it. Should we check it?